src/core/tsi/ssl_transport_security.c

```diff
@@ -862,32 +858,31 @@ static const tsi_frame_protector_vtable frame_protector_vtable = {
 
 static void tsi_ssl_handshaker_factory_destroy(
     tsi_ssl_handshaker_factory *self) {
-  GPR_ASSERT(self != NULL);
-  GPR_ASSERT(self->vtable != NULL);
-  if (self->vtable->destroy != NULL) {
+  if (self == NULL) return;
+
+  if (self->vtable != NULL && self->vtable->destroy != NULL) {
     self->vtable->destroy(self);
   }
   /* Note, we don't free(self) here because this object is always directly
    * embedded in another object. If tsi_ssl_handshaker_factory_init allocates
    * any memory, it should be free'd here. */
 }
 
-static void tsi_ssl_handshaker_factory_addref(
+static tsi_ssl_handshaker_factory *tsi_ssl_handshaker_factory_addref(
```

Comment:
can you change to `_ref` as opposed to `_addref` to be consistent with the code base?